integration/nemo/examples/supervised_fine_tuning/README.md

```diff
@@ -0,0 +1,190 @@
+```
+
+## Federated learning simulations
+Next, we are using NVFlare's [POC mode](https://nvflare.readthedocs.io/en/main/getting_started.html#setting-up-poc) to simulate 
+each client training on their own dataset locally, centralized training with all three dataset combined, and all three clients training together using the 
+[FedAvg](https://arxiv.org/abs/1602.05629) algorithm implemented in NVFlare.
+
+#### 1. Local and Centralized SFT
+For single-site and centralized training experiments, we create the poc workspaces:
+```
+nvflare poc --prepare -n 1
+```
+For 1.3B model experiment, we start the NVFlare system with one GPU:
+```
+nvflare poc --start --gpu 0
+```
+For better usability, open a new terminal and start the [admin command prompt](https://nvflare.readthedocs.io/en/main/real_world_fl/operation.html#admin-command-prompt):
+```
+nvflare poc --start --package admin
+```
+
+We create the configuration files and modify them to include the current directory path to access the dataset and pre-trained LLM.
+At this point, we also modify the data path and local number of clients.
+
+In a standard terminal, run
+```
+python utils/create_configs.py --job_folder "jobs/gpt_sft_1.3B_alpaca" --num_clients 1 --devices 1 --train_ds_files /workspace/Data/Processed/alpaca/training.jsonl --validation_ds_files /workspace/Data/Processed/combined/validation.jsonl
+python utils/create_configs.py --job_folder "jobs/gpt_sft_1.3B_dolly" --num_clients 1 --devices 1 --train_ds_files /workspace/Data/Processed/dolly/training.jsonl --validation_ds_files /workspace/Data/Processed/combined/validation.jsonl
+python utils/create_configs.py --job_folder "jobs/gpt_sft_1.3B_oasst1" --num_clients 1 --devices 1 --train_ds_files /workspace/Data/Processed/oasst1/training.jsonl --validation_ds_files /workspace/Data/Processed/combined/validation.jsonl
+python utils/create_configs.py --job_folder "jobs/gpt_sft_1.3B_combined" --num_clients 1 --devices 1 --train_ds_files /workspace/Data/Processed/combined/training.jsonl --validation_ds_files /workspace/Data/Processed/combined/validation.jsonl
+```
+Note that we used the combined validation set for all experiments, allowing for a direct comparison across all training settings.
+
+Next, submit the single-client SFT jobs using the admin prompt.
+Replace `[PWD]` with the path to this directory.
+```
+submit_job [PWD]/jobs/gpt_sft_1.3B_alpaca
+submit_job [PWD]/jobs/gpt_sft_1.3B_dolly
+submit_job [PWD]/jobs/gpt_sft_1.3B_oasst1
+submit_job [PWD]/jobs/gpt_sft_1.3B_combined
+```
+
+#### 2. Federated SFT
+We use the [FedAvg](https://arxiv.org/abs/1602.05629) algorithm to perform SFT on the model in a federated scenario with 3 clients, each uses one of the three datasets. 
+
+We create the poc workspaces:
+```
+nvflare poc --prepare -n 3
+```
+For 1.3B model experiment, we start the NVFlare system with two GPUs:
```

Comment:
Maybe add a note that the three clients will be distributed over two gpus.

Reply:
clarified this part